src/linker/Linker.Steps/OutputStep.cs

```diff
@@ -103,7 +103,32 @@ protected virtual void WriteAssembly (AssemblyDefinition assembly, string direct
 				}
 			}
 
-			assembly.Write (GetAssemblyFileName (assembly, directory), writerParameters);
+			writerParameters.DeterministicMvid = true;
+
+			var stream = new MemoryStream ();
+
+			var fileName = GetAssemblyFileName (assembly, directory);
+			assembly.Write (stream, writerParameters);
+
+			//
+			// Check whenever the output exits and has the same guid
+			//
+			var prevGuid = Guid.Empty;
+			if (File.Exists (fileName)) {
+				try {
+					var prevModule = ModuleDefinition.ReadModule (fileName);
+					prevGuid = prevModule.Mvid;
+				} catch {
+				}
+			}
+			if (assembly.MainModule.Mvid != prevGuid) {
```

Comment:
Could `assembly.MainModule.Mvid` ever return `Guid.Empty`?  If it's plausible, then that would cause problems when `File.Exists` returns false or when the `try` block above encounters an exception.